### ISSUE TITLE
Training on LSUN bedroom dataset.

### ISSUE THREAD

**Issue by Gomdoree** [autonomousvision/projected_gan#6]
First, thank you for sharing great work!

I trained Projected FastGAN in the LSUN bedroom.

However, FID is increasing across training iterations.

The FID and training loss trends are as follows:

![temp](https://user-images.githubusercontent.com/26226977/144360055-11dbb4c8-ae2f-44a9-9a77-7cd8c7dc63f7.png).

Is there anyone who suffers from the same issue?

Thank you!

**Comment by xl-sr** [comment]
Mhm, this is strange. Can you share your training settings?

**Comment by Gomdoree** [comment]
Thank you for your response!

My training settings are as follows:

(1) hardware settings: RTX 3090 * 4
(2) training config: python train.py --outdir=./training-runs/ --cfg=fastgan --data=../bedroom_train_stylegan2_pytorch.zip  --gpus=4 --batch=64 --mirror=1 --snap=50 --batch-gpu=8 --kimg=10000
(3) Python3.9.7 and pytorch1.9.1.post3 are installed in the machine.
(4) I ran the code on the provided PG Python environment.

One thing I suspect is that my training dataset is created from the official StyleGAN2-ada codebase.

Can it have a bad effect on the model?

Thank you!

**Comment by Gomdoree** [comment]
I want to show the generated samples from my machine.

Specifically, I think the 'fakes000000.png' is strange.

The 'fakes_init.png' is as follows:

![init](https://user-images.githubusercontent.com/26226977/144378897-de21636a-8cf9-4d85-9f0c-b42e95bfee16.png)

The 'fakes000000.png' is as follows:

![iter_0](https://user-images.githubusercontent.com/26226977/144378985-ffbc5e52-201b-489a-8b3f-705745239aa1.png)

The 'fakes007056.png' is as follows:

![iter_7056](https://user-images.githubusercontent.com/26226977/144379041-c02f7559-9884-42a7-808d-702b1c9d1953.png)

Thank you!

**Comment by xl-sr** [comment]
the fakes00000.png are fine actually, this is because this generator uses BatchNorm. 
The samples look like clear mode collapse...

I am going to rerun this dataset in a couple of days when I get to it, could you try another LSUN set, eg., church in the mean time?

**Comment by Gomdoree** [comment]
Okay. I will try another dataset.

Thank you!

**Comment by Gomdoree** [comment]
I trained the Projected FastGAN on the LSUN church dataset.

In this experiment, I created the dataset by using this repository's code. 

The model generates realistic textures, however, the samples look like mode collapse at the same time.

The fakes006451.png is as follows:

![temp_fakes006451](https://user-images.githubusercontent.com/26226977/144548253-2aae5b2a-bbbb-4a87-a362-6cb95d283494.png)

The FID results are as follows:

![temp_church](https://user-images.githubusercontent.com/26226977/144547444-211054e5-9a67-4249-84f9-3f5da68b5899.png)

Thank you!

**Comment by xl-sr** [comment]
Good news: I could reproduce your results and also found the problem :smile: I am going to do some more testing and push the changes later today. Thank you for making me aware of it, this is why I like sharing code!

In the meantime you can try the StyleGAN2 backbone, this one should work fine.

**Comment by Gomdoree** [comment]
Okay! Thanks again. I will wait for the changes.

**Comment by EmreKspToken** [comment]
0x88599654047Adf755bc665De772D8459857d55d7

**Comment by xl-sr** [comment]
Pushed the changes, you should give it a try again :)
Below are the curves you should observe now on LSUN church (blue is FastGAN, pink ist StyleGAN2):

![image](https://user-images.githubusercontent.com/29833625/144588930-49b129d5-0381-4032-9f9e-0126a914fd66.png)

Thanks again for making me aware of this issue.

**Comment by anandbhattad** [comment]
Hi! Do you have pretrained models for LSUN bedrooms trained on StyleGAN2/StyleGAN3? If yes, can you please provide us with those pretrained models? The current one on the webpage is for FastGAN. Thanks for your time and help!
